styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java

```diff
@@ -152,7 +209,95 @@ public void poll(FlyteExecutionId flyteExecutionId, RunState runState)
     } catch (Exception e) {
       throw new PollingException(flyteExecutionId, e);
     }
+  }
+
+  void init() {
+    scheduleWithJitter(this::terminateDanglingFlyteExecutions, scheduledExecutor, terminateDanglingFlyteExecInterval);
+  }
+
+  @VisibleForTesting
+  void terminateDanglingFlyteExecutions() {
+    try {
+      // TODO: add tracing
+      tryTerminateDanglingFlyteExecutions();
+    } catch (Throwable t) {
+      LOG.warn("Error while terminating dangling flyte executions", t);
+    }
+
+  }
+
+  private void tryTerminateDanglingFlyteExecutions() {
+    var projects = flyteAdminClient.listProjects();
+
+    var allFutures = new ArrayList<CompletableFuture<Void>>();
+    for (var project : projects.getProjectsList()) {
+      for (var domain : project.getDomainsList()) {
+        String paginationToken = null;
+        do {
+          var executions =
+              flyteAdminClient.listExecutions(project.getId(), domain.getId(), 100, paginationToken, "");
```

Comment:
do we care about all projects and *all* domains equally?

Reply:
:shrug:  What is your idea? doing production first, then staging and next development? Or are we sure that styx will only execute launch plans in production? SO far we are not enforcing that (I think)

Reply:
yes, let's keep it like that for now. until we something comes up. It's good to avoid "early optimisation" i guess